R/calcHDDCDD.R

```diff
@@ -1,101 +1,714 @@
+                        x2 <- .tlim + 4*tlim_std
+                        y1 <- min(.tlim - 3*tlim_std, tamb - 3*tlim_std)
+                        y2 <- max(.tlim + 3*tlim_std, tamb + 3*tlim_std)
+
+                        if (typeDD == "HDD") {
+                          f <- pracma::integral2(heatingFactor,
+                                                 xmin = x1,
+                                                 xmax = x2,
+                                                 ymin = y1,
+                                                 ymax = function(x){x},
+                                                 tamb = tamb,
+                                                 tamb_std = tamb_std,
+                                                 tlim = .tlim,
+                                                 tlim_std = tlim_std,
+                                                 reltol = 1e-1
+                          )
+                        }
+                        else {
+                          f <- pracma::integral2(coolingFactor,
+                                                 xmin = x1,
+                                                 xmax = x2,
+                                                 ymin = function(x){x},
+                                                 ymax = y2,
+                                                 tamb = tamb,
+                                                 tamb_std = tamb_std,
+                                                 tlim = .tlim,
+                                                 tlim_std = tlim_std,
+                                                 reltol = 1e-1)
+                        }
+                        tmp <- data.frame("T_amb"        = tamb,
+                                          "T_amb_K"      = round(tamb + 273.15, 1),
+                                          "T_lim"        = .tlim,
+                                          "factor"       = f$Q,
+                                          "factor_err"   = f$error,
+                                          "typeDD"       = typeDD)
+                      }
+                    }
+                  )
+                )
+              }
+            )
+          )
+        }
+      )
+    )
+    return(hddcddFactors)
+  }
+
+
+
+  # fill HDD/CDD from factors for given ambient/limit temperature combination
+  calcCellHDDCDD <- function(temp, .typeDD, .tlim, factors) {
+    # extract years
+    dates <- names(temp)
+
+    # add tolerance of 0.04K to avoid machine precision errors
+    factors <- factors %>%
+      filter(.data[["typeDD"]] == .typeDD, .data[["T_lim"]] == .tlim) %>%
+      dplyr::reframe(from = .data[["T_amb_K"]] - 0.049,
+                     to = .data[["T_amb_K"]] + 0.049,
+                     becomes = .data[["factor"]]) %>%
+      data.matrix()
+
+    # swap ambient temperature values with corresponding DD values
+    hddcdd <- classify(temp, factors)
+
+    terra::time(hddcdd) <- as.Date(dates)
```

Comment:
Are you sure you that have another function called `time`? If not, drop the package name.
```suggestion
    time(hddcdd) <- as.Date(dates)
```